### DISCUSSION TITLE
Updating progress tasks from different threads

### DISCUSSION THREAD

**Discussion by akhileshraju** [Textualize/rich#1128] [Q&A] [answered]
Firstly, thanks for the amazing work with Rich! 😀

I am trying to create a `Progress` instance and add tasks to it on the main thread.
After that, I am spinning up 2 more threads to handle the updates for the tasks. This however doesn't work - nothing displays on the console.

Is this something that rich doesn't support OR am I messing something up?

```
import threading
import time

from rich.progress import BarColumn, TextColumn, Progress


def func1(task_id, sleep_time):
    x = 0
    while True:
        time.sleep(sleep_time)
        progress_meter.update(task_id=task_id, status="In Progress", completed=x)
        x += 1

        if x > 100:
            break

    progress_meter.update(task_id=task_id, status="Done")

progress_meter = Progress(
    TextColumn("[progress.description]{task.description}"),
    BarColumn(),
    TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
    TextColumn("{task.fields[status]}"),
)
task_id1 = progress_meter.add_task(
        description=f"Thread 1 progress ",
        status="Starting",
)
task_id2 = progress_meter.add_task(
        description=f"Thread 2 progress ",
        status="Starting",
)
progress_meter.start()

t1 = threading.Thread(target=func1, args=(task_id1, 2))
t2 = threading.Thread(target=func1, args=(task_id2, 3))

t1.start()
t2.start()

t1.join()
t2.join()
```

**Comment by willmcgugan** [comment, marked as answer]
This works for me.

What terminal are you running it on?

**Reply by akhileshraju** [reply]
I was trying it in PyCharm. I ran it in win cmd prompt and it works.
I now rmbr that I had to set `force_terminal=True` for it to show up in the PyCharm run console. I will continue to use that.
I totally frgt that the kind of terminal matters, my bad! Thanks Will 🙂

**Reply by willmcgugan** [reply]
See the note [here](https://rich.readthedocs.io/en/latest/introduction.html) re PyCharm.
